packages/web/e2e/page-object-models/upload.ts

```diff
@@ -0,0 +1,293 @@
+  }
+
+  async openStemsAndDownloadsSettings() {
+    await this.stemsAndDownloadsSettingsButton.click()
+  }
+}
+
+export class EditPlaylistPage extends BaseEditPage {
+  protected readonly titleInput: Locator
+  protected readonly trackList: Locator
+
+  constructor(page: Page) {
+    super(page)
+    this.titleInput = page.getByRole('textbox', { name: /playlist name/i })
+    this.trackList = page.getByRole('list', { name: /track list/i })
+  }
+
+  async setTrackName(index: number, name: string) {
+    const trackItem = this.trackList.getByRole('listitem').nth(index)
+    const nameInput = trackItem.getByRole('textbox', { name: /track name/i })
+    await nameInput.fill(name)
+  }
+
+  async setTrackDetails(
+    index: number,
+    {
+      name,
+      genre,
+      mood,
+      tags
+    }: {
+      name?: string
+      genre?: string
+      mood?: string
+      tags?: string[]
+    }
+  ) {
+    const trackItem = this.trackList.getByRole('listitem').nth(index)
+    const checkbox = trackItem.getByRole('checkbox', {
+      name: /override details for this track/i
+    })
+    if (genre || mood || tags) {
+      await checkbox.check()
+    }
+    if (name) {
+      await this.setTrackName(index, name)
+    }
+    if (genre) {
+      await this.setTrackGenre(index, genre)
+    }
+    if (mood) {
+      await this.setTrackMood(index, mood)
+    }
+    if (tags) {
+      await this.addTrackTags(index, tags)
+    }
+  }
+
+  private async setTrackGenre(index: number, genre: string) {
+    const trackItem = this.trackList.getByRole('listitem').nth(index)
+    const genreBox = trackItem.getByRole('combobox', { name: /pick a genre/i })
+    const genreList = genreBox
+      .page()
+      .getByLabel(/pick a genre/i)
+      .locator('.rc-virtual-list') // ant-d uses a virtualized list
+    await genreBox.fill(genre)
+    await genreList.getByRole('option', { name: genre }).click()
+  }
+
+  private async setTrackMood(index: number, mood: string) {
+    const trackItem = this.trackList.getByRole('listitem').nth(index)
+    const moodBox = trackItem.getByRole('combobox', { name: /pick a mood/i })
+    const moodList = moodBox
+      .page()
+      .getByLabel(/pick a mood/i)
+      .locator('.rc-virtual-list') // ant-d uses a virtualized list
+    await moodBox.fill(mood)
+    await moodList.getByRole('option', { name: mood }).click()
+  }
+
+  private async addTrackTags(index: number, tags: string[]) {
+    const trackItem = this.trackList.getByRole('listitem').nth(index)
+    const tagsInput = trackItem.getByRole('textbox', { name: /tags/i })
+    for (const tag of tags) {
+      await tagsInput.fill(tag)
+      await tagsInput.press('Enter')
+    }
+  }
+}
+
+export class EditAlbumPage extends EditPlaylistPage {
+  protected readonly titleInput: Locator
+
+  constructor(page: Page) {
+    super(page)
+    this.titleInput = page.getByRole('textbox', { name: /album name/i })
+  }
+}
+
+export class SelectPage {
```

Comment:
nit: I realize there's a bit of context given that this is in the upload file, but I still think I would prefer a little more detail in the name
i.e.
```suggestion
export class SelectTracksPage {
```

Reply:
def fair - wanted to keep these names in sync with the ones we have in client but I don't feel too strongly